backend-rust/src/graphql_api.rs

```diff
@@ -2323,31 +2337,66 @@ struct CollectionSegmentInfo {
 }
 
 #[derive(SimpleObject)]
-struct AccountReward {
-    block:       Block,
-    id:          types::ID,
-    timestamp:   DateTime,
-    reward_type: RewardType,
-    amount:      Amount,
+#[graphql(complex)]
+pub struct AccountReward {
+    #[graphql(skip)]
+    id:           i64,
+    #[graphql(skip)]
+    block_height: BlockHeight,
+    timestamp:    DateTime,
+    reward_type:  RewardType,
+    amount:       Amount,
 }
+#[ComplexObject]
+impl AccountReward {
+    async fn id(&self) -> types::ID { types::ID::from(self.id) }
 
-#[derive(Enum, Copy, Clone, PartialEq, Eq)]
-#[allow(clippy::enum_variant_names)]
-enum RewardType {
+    async fn block(&self, ctx: &Context<'_>) -> ApiResult<Block> {
+        Block::query_by_height(get_pool(ctx)?, self.block_height).await
+    }
+}
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq, sqlx::Type)]
+#[sqlx(type_name = "reward_type")]
+pub enum RewardType {
```

Comment:
There should a relationship between RewardTypes and AccountStatementEntries.

However I do not have any immediate good solutions here. As it is of less importance because it only relates to future maintainability I prefer to leave it be.